---
title: "Introduction"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction_SNPVisualizR}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```
## Introduction

`SNPVisualizR` is an R package developed for the integration analysis and visualization of genetic variation data from vcf files.

To download **SNPVisualizR** please use the following commands:

```{r eval=FALSE}
require("devtools")
install_github("terrence-a/SNPVisualizR", build_vignettes = TRUE)
library(SNPVisualizR)
```

The following command lists all the sample functions available as a part of this package.

``` {r eval=FALSE}
ls("package:SNPVisualizR")
```

##Components
###Available Functions
1. **loadVCF**
The **loadVCF* function is responsible for creating an R object which contains the genetic variation data from the vcf file you specify:

```{r eval=FALSE}
vcfDat <- loadVCF("ex.vcf")
```

The above function will create an S4 object with the attributes `species`, `contifInfo`, and `variantData`. These attributes will be set to values corresponding to the data from the specified vcf file.
**loadVCF** also accepts a `filesizelimit` argument, which can be used to select the first n line of vcf (to avoid having to wait for an entire vcf file to be read (default = $1^6$))

2. **loadFA**
The **loadFA** function is responsible for creating an R object which contains sequence data from a specified fasta file.

```{r eval=FALSE}
faDat <- loadFA("ex.fa")
```
3. **plotPosition**
The **plotPosition** function creates a plot of the variant data, plotting the frequencies of SNPs along a fixed position axis.
The **plotPosition** requires only a vcfObject S4 Object (created by `loadVCF`) to plot.
```{r eval=FALSE}
  plotPosition(vcfDat)
```

4. **calculateTiTv**
The **calculateTiTv** function calculates the Transition/Transversion ratio of all the listed variants in the file.

```{r eval=FALSE}
calculateTiTv(vcfDat)
```
